File: Demo-LLP-MIP/Demo-LLP-MIP-Python/v1/cashier/argument/create/CreatePayParams.py
from v1.info.GoodsInfo import GoodsInfo
from v1.info.DeviceInfo import DeviceInfo
from v1.info.PayMethodInfo import PayMethodInfo
from v1.info.DivideInfo import DivideInfo
from v1.info.StoreInfo import StoreInfo
from v1.info.RiskInfo import RiskInfo
from v1.info.Style import Style

class CreatePayParams:
    def __init__(self):
        self.mch_id = None
        self.sub_mchid = None
        self.user_id = None
        self.busi_type = None
        self.txn_seqno = None
        self.txn_time = None
        self.order_amount = None
        self.order_info = None
        self.pay_expire = None
        self.notify_url = None
        self.return_url = None
        self.share_flag = None
        self.secured_flag = None
        self.goods_info = None
        self.risk_info = None  # 假设 RiskInfo 为另一个类
        self.device_info = None  # 假设 DeviceInfo 为另一个类
        self.store_info = None  # 假设 StoreInfo 为另一个类
        self.payee_infos = None
        self.style = None  # 假设 Style 为另一个类
        self.extend_info = None
    def to_dict(self):
        data= {
            "mch_id": self.mch_id,
            "sub_mchid": self.sub_mchid,
            "user_id": self.user_id,
            "busi_type": self.busi_type,
            "txn_seqno": self.txn_seqno,
            "txn_time": self.txn_time,
            "order_amount": self.order_amount,
            "order_info": self.order_info,
            "pay_expire": self.pay_expire,
            "notify_url": self.notify_url,
            "return_url": self.return_url,
            "share_flag": self.share_flag,
            "secured_flag": self.secured_flag,
            # "goods_info": [goods.__dict__ for goods in self.goods_info],
            # "risk_info": self.risk_info.__dict__,
            # "device_info": self.device_info.__dict__,
            # "store_info": self.store_info.__dict__,
            # "payee_infos": [payee.__dict__ for payee in self.payee_infos],
            # "style": self.style.__dict__,
            "extend_info": self.extend_info
        }
        if self.goods_info is not None:
            data["goods_info"] = [goods.__dict__ for goods in self.goods_info]
        if self.risk_info is not None:
            data["risk_info"] = self.risk_info.__dict__
        if self.device_info is not None:
            data["device_info"] = self.device_info.__dict__
        if self.store_info is not None:
            data["store_info"] = self.store_info.__dict__
        if self.payee_infos is not None:
            data["payee_infos"] = [payee.__dict__ for payee in self.payee_infos]
        if self.style is not None:
            data["style"] = self.style.__dict__

        data = {k: v for k, v in data.items() if v is not None}
        return {k: v for k, v in data.items() if v}

File: Demo-LLP-MIP/Demo-LLP-MIP-PHP/v1/cashier/argument/create/CreatePayParams.php
<?php


class CreatePayParams {
    public $mch_id;
    public $sub_mchid;
    public $user_id;
    public $busi_type;
    public $txn_seqno;
    public $txn_time;
    public $order_amount;
    public $order_info;
    public $pay_expire;
    public $notify_url;
    public $return_url;
    public $share_flag;
    public $secured_flag;
    public $goods_info;
    public $risk_info;
    public $device_info;
    public $store_info;
    public $payee_infos;
    public $style;
    public $extend_info;

    public function to_dict() {
        $data = [];
        foreach ($this as $key => $value) {
            if ($value !== null) {
                $data[$key] = $value;
            }
        }
        return json_encode($data);
    }

//    public function to_dict() {
//        $data = array(
//            "mch_id" => $this->mch_id,
//            "sub_mchid" => $this->sub_mchid,
//            "user_id" => $this->user_id,
//            "busi_type" => $this->busi_type,
//            "txn_seqno" => $this->txn_seqno,
//            "txn_time" => $this->txn_time,
//            "order_amount" => $this->order_amount,
//            "order_info" => $this->order_info,
//            "pay_expire" => $this->pay_expire,
//            "notify_url" => $this->notify_url,
//            "return_url" => $this->return_url,
//            "share_flag" => $this->share_flag,
//            "secured_flag" => $this->secured_flag,
//            "extend_info" => $this->extend_info
//        );
//
//        if (!empty($this->goods_info)) {
//            $data["goods_info"] = array_map(function($goods) {
//                return $goods->to_dict();
//            }, $this->goods_info);
//        }
//        if (!empty($this->risk_info)) {
//            $data["risk_info"] = $this->risk_info->to_dict();
//        }
//        if (!empty($this->device_info)) {
//            $data["device_info"] = $this->device_info->to_dict();
//        }
//        if (!empty($this->store_info)) {
//            $data["store_info"] = $this->store_info->to_dict();
//        }
//        if (!empty($this->payee_infos)) {
//            $data["payee_infos"] = array_map(function($payee) {
//                return $payee->to_dict();
//            }, $this->payee_infos);
//        }
//        if (!empty($this->style)) {
//            $data["style"] = $this->style->to_dict();
//        }
//
//        $data = array_filter($data, function($value) {
//            return !is_null($value);
//        });
//        return array_filter($data, function($value) {
//            return $value;
//        });
//    }
}
?>